wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/Reactions/ReactionCollectionView.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - ReactionCollectionView
+
+final class ReactionCollectionView: UIView, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout, UICollectionViewDelegate {
+
+    // MARK: - Properties
+
+    private var collectionViewHeight: CGFloat = 40
+    lazy var collectionViewHeightConstraint: NSLayoutConstraint = collectionView.heightAnchor.constraint(equalToConstant: collectionViewHeight)
+    let flowLayout = UICollectionViewFlowLayout()
+    private var contentSizeObservation: NSKeyValueObservation?
+    private lazy var collectionView: UICollectionView = {
+        return UICollectionView(frame: .zero, collectionViewLayout: self.flowLayout)
+    }()
+
+    var reactions = [MessageReactionMetadata]() {
```

Comment:
Can this be private ? How do we make sure it's called on mainThread, see didSet reloadData()

Reply:
No, it can't be private since it's being used on MessageReactionsCell.

```swift
func configure(
        with object: Configuration,
        animated: Bool
    ) {
        reactionCollectionView.reactions = object.reactions.map {
            var reaction = $0

            reaction.performReaction = { [weak self] in
                guard
                    let `self` = self,
                    let message = self.message
                else {
                    return
                }

                self.delegate?.perform(
                    action: .react(reaction.type),
                    for: message,
                    view: self
                )
            }

            return reaction
        }

    }
```

Reply:
just make sure then that it's always called on mainThread or add a DispatchQueue.main.async in the setter.

Reply:
Done!